python/ql/src/semmle/python/frameworks/Stdlib.qll

```diff
@@ -1032,6 +1032,538 @@ private module Stdlib {
 
     override string getFormat() { result = "JSON" }
   }
+
+  // ---------------------------------------------------------------------------
+  // cgi
+  // ---------------------------------------------------------------------------
+  /** Gets a reference to the `cgi` module. */
+  private DataFlow::Node cgi(DataFlow::TypeTracker t) {
+    t.start() and
+    result = DataFlow::importNode("cgi")
+    or
+    exists(DataFlow::TypeTracker t2 | result = cgi(t2).track(t2, t))
+  }
+
+  /** Gets a reference to the `cgi` module. */
+  DataFlow::Node cgi() { result = cgi(DataFlow::TypeTracker::end()) }
+
+  /** Provides models for the `cgi` module. */
+  module cgi {
+    /**
+     * Provides models for the `cgi.FieldStorage` class
+     *
+     * See https://docs.python.org/3/library/cgi.html.
+     */
+    module FieldStorage {
+      /** Gets a reference to the `cgi.FieldStorage` class. */
+      private DataFlow::Node classRef(DataFlow::TypeTracker t) {
+        t.startInAttr("FieldStorage") and
+        result = cgi()
+        or
+        exists(DataFlow::TypeTracker t2 | result = classRef(t2).track(t2, t))
+      }
+
+      /** Gets a reference to the `cgi.FieldStorage` class. */
+      DataFlow::Node classRef() { result = classRef(DataFlow::TypeTracker::end()) }
+
+      /**
+       * A source of an instance of `cgi.FieldStorage`.
+       *
+       * This can include instantiation of the class, return value from function
+       * calls, or a special parameter that will be set when functions are call by external
+       * library.
+       *
+       * Use `FieldStorage::instance()` predicate to get references to instances of `cgi.FieldStorage`.
```

Comment:
```suggestion
       * An instance of `cgi.FieldStorage`, extend this class to model new instances.
       *
       * This can include instantiations of the class, return values from function
       * calls, or a special parameter that will be set when functions are called by an external
       * library.
       *
       * Use the predicate `FieldStorage::instance()` to get references to instances of `cgi.FieldStorage`.
```

Reply:
```diff
-A source of an instance of `cgi.FieldStorage`.
+An instance of `cgi.FieldStorage`
```

I don't agree with this part. This pattern of having a source of an instance doesn't make too much sense here, but really helps out in other places, for example for the `MultiDict` in werkzeug. I would really like us to not change this pattern, so it's easy to recognize it's just this _one_ pattern being used all over the place.

The other suggestions looks like an improvement. Let's do that in an other PR?

Reply:
Happy to relegate these to a separate PR if we are planning to change multiple instances anyway.

Reply:
Perhaps "A source of instances of...", then? I wanted to make it clear, that if you extend this class, you should extend it with things that can be considered instances since the code below does so. The "source of" part was confusing this a bit for me.

Reply:
> A source of instances of...

sounds good to me :+1: